components/automate-ui/src/app/services/chef-session/chef-session.service.ts

```diff
@@ -61,36 +70,45 @@ export class ChefSessionService implements CanActivate {
       // and components/session-service and chef-session-service -- it would
       // make sense to try to refresh the session if we get a 401 from the API,
       // before giving up and calling logout().
-      window.setInterval(() => {
-        /* Using XMLHttpRequest instead of angular/http to avoid
-           dependency hell. */
-        const xhr = new XMLHttpRequest();
-        xhr.onreadystatechange = this.refreshSessionCallback.bind(this);
-        xhr.open('GET', '/session/refresh', true);
-        xhr.responseType = 'json';
-        xhr.setRequestHeader('Authorization', `Bearer ${this.id_token}`);
-        xhr.send(null);
-      }, minute);
+      timer(0, minute).pipe(
+        filter(_ => !this.isRefreshing),
+        mergeMap(_ => {
+          this.isRefreshing = true;
+          return this.refresh();
+        }),
+      ).subscribe(
+        token => {
+          this.ingestIDToken(token);
+          this.isRefreshing = false;
+        },
+        error => {
+          this.isRefreshing = false;
+          if (error instanceof HttpErrorResponse) {
+            if (error.status == HTTP_STATUS_UNAUTHORIZED) {
+              this.logout();
+            } else {
+              console.log(`Session refresh failed: ${error.statusText}`);
```

Comment:
Wondering if we want to expose these more to the user with the notification banner? 🤷

Reply:
Isn't this going to become the new "could not verify license"? I.e. the thing that'll happen when the backend is not responding? I think we should try to verify that behaviour. If it's only logging to the console now, we should know.